pr_agent/algo/ai_handler.py

```diff
@@ -24,7 +26,7 @@ def __init__(self):
         Raises a ValueError if the OpenAI key is missing.
         """
         self.azure = False
-
+        self.api_base = None
         if get_settings().get("OPENAI.KEY", None):
```

Comment:
**Suggestion:** Consider using a configuration object or dictionary to store all the settings instead of calling `get_settings()` multiple times. This can improve performance by reducing the number of function calls. [performance]
```suggestion
        settings = get_settings()
        if settings.get("OPENAI.KEY", None):
            openai.api_key = settings.openai.key
            litellm.openai_key = settings.openai.key
        if settings.get("HUGGINGFACE.KEY", None):
            litellm.huggingface_key = settings.huggingface.key
        if settings.get("HUGGINGFACE.API_BASE", None) and 'huggingface' in settings.config.model:
            litellm.api_base = settings.huggingface.api_base
            self.api_base = settings.huggingface.api_base
        if settings.get("VERTEXAI.VERTEX_PROJECT", None):
            litellm.vertex_project = settings.vertexai.vertex_project
            litellm.vertex_location = settings.get(
```